### PR TITLE
chore: Extract record builder functionality from HandleContext

### PR DESCRIPTION
**Pull request by netopyr**
**Description**:

This PR extracts the record builder functionality from HandleContext

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     * Returns a record builder for the given record builder subtype.
+     *
+     * @param recordBuilderClass the record type
+     * @param <T> the record type
+     * @return a builder for the given record type
+     * @throws NullPointerException if {@code recordBuilderClass} is {@code null}
+     * @throws IllegalArgumentException if the record builder type is unknown to the app
+     */
+    @NonNull
+    <T> T current(@NonNull Class<T> recordBuilderClass);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:35]: nit : May be can be named as `getOrCreate()`. So it looks like `context.recordBuilders().getOrCreate(TokenAccountWipeRecordBuilder.class)`?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM other than one comment. Thanks @netopyr

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:35]: Great idea! I will change the name.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM! Thanks @netopyr

**Review by Neeharika-Sompalli — APPROVED** [review]

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+     * Returns a record builder for the given record builder subtype.
+     *
+     * @param recordBuilderClass the record type
+     * @param <T> the record type
+     * @return a builder for the given record type
+     * @throws NullPointerException if {@code recordBuilderClass} is {@code null}
+     * @throws IllegalArgumentException if the record builder type is unknown to the app
+     */
+    @NonNull
+    <T> T getOrCreate(@NonNull Class<T> recordBuilderClass);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:35]: Should these arguments be `final`? (all methods)

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     * @return the new child record builder
+     * @param <T> the record type
+     * @throws NullPointerException if {@code recordBuilderClass} is {@code null}
+     * @throws IllegalArgumentException if the record builder type is unknown to the app
+     */
+    @NonNull
+    <T> T addChildRecordBuilder(@NonNull Class<T> recordBuilderClass);
+
+    /**
+     * Adds a removable child record builder to the list of record builders. Unlike a regular child record builder,
```
Comment (david-bakin-sl) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:51]: Reading this comment and the comment above at `addChildRecordBuilder` I don't get the difference between them, unless somehow it is that one is "removed" and the other "reverted", but in that case I don't understand the terminology.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java (hunk 2 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:35]: No, `RecordBuilders` is an interface.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java (hunk 3 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:51]: In one case, the child record builder is removed from the list if the parent transaction fails, meaning the child record will not appear in the stream. In the other case, it is reverted if the parent transaction fails, meaning there will be a record in the record stream, but it has the status `REVERTED_SUCCESS`.

Comment (david-bakin-sl) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:51]: TY for the explanation.  But ... can it be memorialized in the comment?

Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:51]: Sure, I can do it in a follow-up PR.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java (hunk 2 above)
Comment (david-bakin-sl) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/records/RecordBuilders.java:35]: ok.

**Review by david-bakin-sl — APPROVED** [review]
LGTM!

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**